pkg/rebuild/rebuild/git_test.go

```diff
@@ -0,0 +1,125 @@
+package rebuild
+
+import (
+	"testing"
+
+	"github.com/go-git/go-billy/v5/memfs"
+	git "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestMatchTag(t *testing.T) {
+	tests := []struct {
+		tag     string
+		pkg     string
+		version string
+		strict  bool
+		approx  bool
+	}{
+		{"v1.0.0", "mypackage", "1.0.0", true, true},
+		{"v1.0.0-rc1", "mypackage", "1.0.0", false, true},
+		{"mypackage-1.0.0", "mypackage", "1.0.0", true, true},
+		{"1.0.0", "mypackage", "1.0.0", true, true},
+		{"v1.0.1", "mypackage", "1.0.0", false, false},
+		{"v1.0", "mypackage", "1.0.0", false, false},
+		{"v1", "mypackage", "1.0.0", false, false},
+		{"org/mypackage-1.0.0", "org/mypackage", "1.0.0", true, true},
+		{"mypackage-1.0.0", "org/mypackage", "1.0.0", true, true},
+		{"org/otherpackage-1.0.0", "org/mypackage", "1.0.0", false, true}, // org-but-not-package special case
+		{"otherpackage-1.0.0", "org/mypackage", "1.0.0", true, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.tag, func(t *testing.T) {
+			strict, approx := MatchTag(tt.tag, tt.pkg, tt.version)
+			if strict != tt.strict || approx != tt.approx {
+				t.Errorf("MatchTag(%q, %q, %q) = (%v, %v), want (%v, %v)", tt.tag, tt.pkg, tt.version, strict, approx, tt.strict, tt.approx)
+			}
+		})
+	}
+}
+
+func TestFindTagMatch(t *testing.T) {
+	repo := must(git.Init(memory.NewStorage(), memfs.New()))
+
+	c1 := createCommit(repo, "commit1")
+	c2 := createCommit(repo, "commit2")
+	createLightweightTag(repo, "v1.0.0", c1)
+	createLightweightTag(repo, "v1.1.0", c2)
+	createAnnotatedTag(repo, "v1.0.0-alpha", c1)
```

Comment:
Does this need to be on the same c1 for the tests? It would be helpful if this was a separate c3 to make it clear when TagMatch is using the annotated tag vs the lightweight tag.

Reply:
it doesn't need to be. was just briefer. changed